codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientGenerator.kt

```diff
@@ -132,121 +127,73 @@ class FluentClientGenerator(
+                                    Set the `sleep_impl` on the Config passed into this function to fix this panic.");
                         }
-                    }
 
-                    impl${generics.inst} From<#{client}::Client#{smithy_inst:W}> for Client${generics.inst} {
-                        fn from(client: #{client}::Client#{smithy_inst:W}) -> Self {
-                            Self::with_config(client, crate::Config::builder().build())
+                        Self {
+                            handle: #{Arc}::new(
+                                Handle {
+                                    conf: conf.clone(),
+                                    runtime_plugins: #{base_client_runtime_plugins}(conf),
+                                }
+                            )
                         }
                     }
 
-                    impl${generics.inst} Client${generics.inst} {
-                        /// Creates a client with the given service configuration.
-                        pub fn with_config(client: #{client}::Client#{smithy_inst:W}, conf: crate::Config) -> Self {
-                            Self {
-                                handle: #{Arc}::new(Handle {
-                                    client,
-                                    conf,
-                                })
-                            }
-                        }
-
-                        /// Returns the client's configuration.
-                        pub fn conf(&self) -> &crate::Config {
-                            &self.handle.conf
-                        }
-                    }
-                    """,
-                    *clientScope,
-                )
-            } else {
-                rustTemplate(
-                    """
-                    ##[derive(Debug)]
-                    pub(crate) struct Handle {
-                        pub(crate) conf: crate::Config,
-                        pub(crate) runtime_plugins: #{RuntimePlugins},
+                    /// Returns the client's configuration.
+                    pub fn config(&self) -> &crate::Config {
+                        &self.handle.conf
                     }
 
-                    #{client_docs:W}
-                    ##[derive(#{Clone}, ::std::fmt::Debug)]
-                    pub struct Client {
-                        handle: #{Arc}<Handle>,
+                    ##[doc(hidden)]
+                    // TODO(enableNewSmithyRuntimeCleanup): Delete this function when cleaning up middleware
+                    // This is currently kept around so the tests still compile in both modes
+                    /// Creates a client with the given service configuration.
+                    pub fn with_config<C, M, R>(_client: #{client}::Client<C, M, R>, conf: crate::Config) -> Self {
+                        Self::from_conf(conf)
                     }
```

Comment:
We still need this for some tests, right? Same for the `conf` method below?

Reply:
Will address the TODO in a later PR. This method isn't needed anymore.